### PR TITLE
[Chore] Changelog for 0.71.0

### PR DESCRIPTION
**Pull request by cipolleschi**
## Summary

Adds changelog for new release.

## Changelog

[Internal] [Changed] - add changelog entry for 0.71.0

## Test Plan

N/A

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,417 @@
 # Changelog
 
+## v0.71.0-rc.0
```
Comment (kelset) [inline CHANGELOG.md:3]: just a reminder that we need to also add the RC1 commits https://github.com/facebook/react-native/releases/tag/v0.71.0-rc.1

Comment (kelset) [reply CHANGELOG.md:3]: And for RC2: https://github.com/facebook/react-native/releases/tag/v0.71.0-rc.2

Comment (kelset) [reply CHANGELOG.md:3]: And for RC3: https://github.com/facebook/react-native/releases/tag/v0.71.0-rc.3

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,421 @@
...
+- Add role prop to Text component ([20718e6b8c](https://github.com/facebook/react-native/commit/20718e6b8ce4f5c3a8393067d5e8eb0da910751c) by [@gabrieldonadel](https://github.com/gabrieldonadel))
+- Introduce `useAnimatedValue` hook to make it easier working with `Animated.Value`s in function components. ([e22217fe8b](https://github.com/facebook/react-native/commit/e22217fe8b9455e32695f88ca835e11442b0a937) by [@fabriziocucci](https://github.com/fabriziocucci))
+- Highlight elements on hover while mouse down for React DevTools element inspection. ([94429fb037](https://github.com/facebook/react-native/commit/94429fb037322652b6796ac2c40cef451795205e))
+- Implement method bindings for gap/row-gap/column-gap ([803a2978e5](https://github.com/facebook/react-native/commit/803a2978e5c0a33c6bb57b402a67218571382fbb) by [@NickGerleman](https://github.com/NickGerleman))
+- Implement gap/row-gap/column-gap (within the C ABI) ([1373a7057b](https://github.com/facebook/react-native/commit/1373a7057b9d66830868832a9684ad80ae5d9d45) by [@NickGerleman](https://github.com/NickGerleman))
```
Comment (jacobp100) [inline CHANGELOG.md:30]:
```suggestion
- Implement gap/row-gap/column-gap (within the C ABI) ([1373a7057b](https://github.com/facebook/react-native/commit/1373a7057b9d66830868832a9684ad80ae5d9d45) by [@NickGerleman](https://github.com/NickGerleman) and [@jacobp100](https://github.com/jacobp100))
```

Comment (jacobp100) [reply CHANGELOG.md:30]: Think this is just due to how it was committed 😅

**Pull request closed without merging**